woocommerce/woocommerce
# Fix: handle undefined templateSlug in Product Collection tracking utils

### Changes proposed in this Pull Request:





The modification introduces a check for templateSlug before attempting to access its properties or use it in further logic. This change addresses a bug that manifested when users tried to edit Product Collection block in Sync Pattern, resulting in a JavaScript error because templateSlug could be undefined.

Closes #47502 



### How to test the changes in this Pull Request:



1. Go to "Products by category" template & add product collection block to it.
2. Create a sync pattern with Product Collection block inside it.
3. Select sync pattern and click on "Edit original" in toolbar as shown in screenshot below
![image](https://github.com/woocommerce/woocommerce/assets/16707866/44a93edc-8b5f-4bef-af9e-d576fd45450b)


4. Verify that the following error isn't occurring when the Sync pattern is loaded in Editor.
![image](https://github.com/woocommerce/woocommerce/assets/16707866/1386b697-b933-471e-bd62-1b83bcddcbab)

5. Developer only testing: Verify that you are able to reproduce [this issue](https://github.com/woocommerce/woocommerce/issues/47502) on `trunk` but not on this PR branch. 



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Fix: handle undefined templateSlug in Product Collection tracking utils

#### Comment 

</details>
